tensorboard/plugins/image/tf_image_dashboard/tf-image-loader.html

```diff
@@ -316,12 +321,28 @@
         img.src = steps[stepIndex].url;
       },
 
-      _toggleExpanded(e) {
-        this.set('_expanded', !this._expanded);
+      _toggleExpanded() {
+        const newValue = !this._expanded;
+        if (newValue) {
+          // If this is true, _actualSizeShown must be false. The features are mutually exclusive.
+          this.set('_actualSizeShown', false);
+        }
+        this.set('_expanded', newValue);
+      },
+      _toggleActualSizeShown() {
+        const newValue = !this._actualSizeShown;
+        if (newValue) {
+          // If this is true, expanded must be false. The features are mutually exclusive.
+          this.set('_expanded', false);
+        }
+        this.set('_actualSizeShown', newValue);
       },
-      _showActualSize() {
-        const img = this.$$("#main-image-container img");
-        this.showActualSize(img);
+      _updateActualImageSize(actualSizeShown) {
+        // Updating the attribute and then relying on a descendent selector, ie
+        // ":host[_actualSizeShown]" img does not work, so we manually update the image width
```

Comment:
Add `reflectToAttribute: true,  // for CSS` in the definition of the `_actualSizeShown` property, and then it will work. See `_expanded` for an example. (Then we can get rid of this function, increasing the declarative:imperative ratio.)

I suspect that the selector that you want is `:host[_actualSizeShown] #main-image-container img`.